index.js

```diff
@@ -169,7 +169,91 @@ class Pawn extends Piece {
   get icon() {
     return this.isWhite() ? "♙" : "♟";
   }
+  onMove(move) {
+    this.hasMoved = true;
+  }
+}
+
+class Rook extends Piece {
+  constructor(game, board, color, row, file) {
+    super(game, board, color);
+    this.row = row;
+    this.file = file;
```

Comment:
Since every subclass is going to have a `row` and a `file` property, let's move these into the parent class.
 ```suggestion
    super(game, board, color, row, file);
```